docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+======================
+Full Stack Application
+======================
+
+Overview
+========
+In this tutorial, we’ll walk you through using Smithy in a full-stack application. This includes defining a model for
+a simple coffee-shop service, generating TypeScript code for both the client and server, and implementing both a
+front-end and back-end for the service. We won't assume that you're an expert in Smithy, but it may helpful to work
+through the :doc:`../quickstart` before beginning this tutorial.
+
+Let's get started.
+
+Setting Up the Project
+======================
+This application will consist of a 4 major components:
+
+1. The model
+2. The server
+3. The client
+4. The web application
+
+--------------
+Pre-requisites
+--------------
+To follow this tutorial, you'll need to install a few tools.
+
+* :doc:`Smithy CLI <../guides/smithy-cli/cli_installation>`
+* `Node.js (>= 16) <https://nodejs.org/en/download>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_
+
+------
+Set up
+------
+Once you have these installed, the CLI has a useful command to easily bootstrap projects from
+`the repository of example smithy projects <https://github.com/smithy-lang/smithy-examples>`_. Execute the following
+command to set up the initial project:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    smithy init -t full-stack-application
+
+You will now have the project in the ``full-stack-application`` directory. In this project, you will find a ``README``
+that contains important information about this project, like how it is laid out and how to build and run it. In this
+tutorial, we will show exactly which commands to run and when.
+
+.. TODO: Provide the skeleton template or a git patch?
+
+Modeling Our Service
```

Comment:
Maybe our "Coffee Shop" Service. I think it helps to re-iterate what we are building.